R/fixed_design_ahr.R

```diff
@@ -0,0 +1,121 @@
+#  Copyright (c) 2023 Merck & Co., Inc., Rahway, NJ, USA and its affiliates.
+#  All rights reserved.
+#
+#  This file is part of the gsDesign2 program.
+#
+#  gsDesign2 is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' Fixed design sample size
+#'
+#' Computes fixed design sample size for AHR methods.
+#' Returns a tibble with a basic summary.
+#'
+#' @inheritParams gs_design_ahr
+#' @inheritParams gs_power_ahr
+#' @param power Power (`NULL` to compute power or strictly between 0
+#'   and `1 - alpha` otherwise).
+#' @param study_duration Study duration.
+#'
+#' @return A table.
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' # example 1: given power and compute sample size
+#' x <- fixed_design_ahr(
+#'   alpha = .025, power = .9,
+#'   enroll_rate = define_enroll_rate(duration = 18, rate = 1),
+#'   fail_rate = define_fail_rate(
+#'     duration = c(4, 100),
+#'     fail_rate = log(2) / 12,
+#'     hr = c(1, .6),
+#'     dropout_rate = .001
+#'   ),
+#'   study_duration = 36
+#' )
+#' x %>% summary()
+#' # example 2: given sample size and compute power
+#' x <- fixed_design_ahr(
+#'   alpha = .025,
+#'   enroll_rate = define_enroll_rate(duration = 18, rate = 20),
+#'   fail_rate = define_fail_rate(
+#'     duration = c(4, 100),
+#'     fail_rate = log(2) / 12,
+#'     hr = c(1, .6),
+#'     dropout_rate = .001
+#'   ),
+#'   study_duration = 36
+#' )
+#' x %>% summary()
+fixed_design_ahr <- function(alpha = 0.025,
+                             power = NULL,
+                             ratio = 1,
+                             study_duration = 36,
+                             enroll_rate,
```

Comment:
please put `enroll_rate` and `fail_rate` as first two arguments.

Reply:
Just updated it in the latest commit.